mayastor/src/grpc/controller_grpc.rs

```diff
@@ -1,132 +1,72 @@
 use crate::{
-    bdev::{nexus, NvmeController, NvmeControllerState, NVME_CONTROLLERS},
+    bdev::{NvmeController, NvmeControllerState, NVME_CONTROLLERS},
     core::{BlockDeviceIoStats, CoreError},
     ffihelper::{cb_arg, done_cb},
-    grpc::{rpc_submit, GrpcResult},
 };
-
-use ::rpc::mayastor as rpc;
 use futures::channel::oneshot;
-use std::convert::From;
-use tonic::{Response, Status};
+
+#[derive(Debug)]
+pub struct NvmeControllerInfo {
+    pub name: String,
+    pub state: NvmeControllerState,
+    pub size: u64,
+    pub blk_size: u32,
+}
 
 impl<'a> NvmeController<'a> {
-    fn to_grpc(&self) -> rpc::NvmeController {
+    fn to_info(&self) -> NvmeControllerInfo {
         let (size, blk_size) = self
             .namespace()
             .map_or((0, 0), |ns| (ns.size_in_bytes(), ns.block_len() as u32));
 
-        rpc::NvmeController {
+        NvmeControllerInfo {
             name: self.name.to_string(),
-            state: rpc::NvmeControllerState::from(self.get_state()) as i32,
+            state: self.get_state(),
             size,
             blk_size,
         }
     }
 }
 
-impl From<NvmeControllerState> for rpc::NvmeControllerState {
-    fn from(state: NvmeControllerState) -> Self {
-        match state {
-            NvmeControllerState::New => rpc::NvmeControllerState::New,
-            NvmeControllerState::Initializing => {
-                rpc::NvmeControllerState::Initializing
-            }
-            NvmeControllerState::Running => rpc::NvmeControllerState::Running,
-            NvmeControllerState::Faulted(_) => {
-                rpc::NvmeControllerState::Faulted
-            }
-            NvmeControllerState::Unconfiguring => {
-                rpc::NvmeControllerState::Unconfiguring
-            }
-            NvmeControllerState::Unconfigured => {
-                rpc::NvmeControllerState::Unconfigured
+/// Returns the IO stats of the given NVMe Controller
+pub async fn controller_stats(
+    controller_name: &str,
+) -> Result<BlockDeviceIoStats, CoreError> {
```

Comment:
I might have overlooked it, but could you please remind why we return here 'BlockDeviceIoStats' instead of returning 'NvmeControllerIoStats' directly ? All callers of 'controller_stats()' aim to end up with 'NvmeControllerIoStats', nobody uses raw 'BlockDeviceIoStats'.

```
let stats = controller_stats(&ctrl.name).await;
                        if stats.is_ok() {
                            res.push(NvmeControllerStats {
                                name: ctrl.name,
                                stats: stats
                                    .ok()
                                    .map(NvmeControllerIoStats::from),
                            });
                        }
```

Could we make this function return 'NvmeControllerIoStats' directly, so that we can eliminate unnecessary '.map()' (unless I overlooked some other use-cases of this function that require 'BlockDeviceIoStats' ?
Probably this will also not needed then:

```
 'impl From<BlockDeviceIoStats> for rpc::NvmeControllerIoStats {'
```

Reply:
The change was done so that we dont have to use the rpc::NvmeControllerIoStats directly. (since we are creating a v1 version.)  

What we can do is create an `NvmeControllerIoStats` similar to the `NvmeControllerInfo` defined in this same file and then convert it to the rpc::NvmeControllerIoStats. This will avoid passing the `BlockDeviceIoStats` as is. This struct will have the exact same fields of  the `BlockDeviceIoStats`, thats why used it directly.

Reply:
Thanks for the explanation. I think we're good now.